examples/pdf-bookmarks.cc

```diff
@@ -138,20 +139,14 @@ void show_bookmark_details(QPDFOutlineObjectHelper outline,
 void extract_bookmarks(std::vector<QPDFOutlineObjectHelper> outlines,
                        std::vector<int>& numbers)
 {
-    numbers.push_back(0);
-    for (std::vector<QPDFOutlineObjectHelper>::iterator iter = outlines.begin();
-         iter != outlines.end(); ++iter)
+    // for style == st_lines : numbers.at(n) == 0 => last outline at level n
```

Comment:
Does:

// for (style == st_lines), we count 'numbers.at(n)' up to 0 to identify the last bookmark (at level n).

work for you.

Reply:
Yeah, that's fine, though I still didn't understand it until I read through the whole code. Maybe
```
    // For style == st_numbers, numbers.at(n) contains the numerical
    // label for the outline, so we count up from 1.
    // For style == st_lines, numbers.at(n) == 0 indicates the last
    // outline at level n, and we don't otherwise care what the value
    // is, so we count up to zero.
```